mmv1/third_party/terraform/services/compute/compute_instance_helpers.go.erb

```diff
@@ -272,6 +275,7 @@ func flattenScheduling(resp *compute.Scheduling) []map[string]interface{} {
 		"min_node_cpus":       resp.MinNodeCpus,
 		"provisioning_model":  resp.ProvisioningModel,
 		"instance_termination_action": resp.InstanceTerminationAction,
+		"availability_domain": resp.AvailabilityDomain,
```

Comment:
We're writing `availability_domain` to instance template due to the shared code, but the value isn't available. I assume the unset value is `nil` so we can't just use a conditional write, an approach used in some other places. We may want to add the field to schema in instance template and add a note it doesn't currently work (but that could lead to weird behaviour if it's added in the future).

Otherwise we may need to pass a flag through from the initial callsite about whether we're working on a template or actual instance, and use that to control what we write.

(I would personally like to remove the shared code eventually to avoid these kind of decisions, not that maintaining separate copies will be very fun 🙁 )